src/lib/server/loadHomeData.ts

```diff
@@ -46,84 +41,131 @@ export const loadHomeData = async ({
 
   // EVENTS
-  const eventsPromise = prisma.event
-    .findMany({
-      where: {
-        startDatetime: {
-          gt: new Date(),
-        },
-      },
-      orderBy: {
-        startDatetime: "asc",
-      },
-      take: 3,
-    })
-    .then((events) =>
-      events.map((event) => ({
-        ...event,
-        description: DOMPurify.sanitize(event.description),
-        descriptionEn: event.descriptionEn
-          ? DOMPurify.sanitize(event.descriptionEn)
-          : event.descriptionEn,
-      })),
-    );
+  const eventsPromise = userLevelCached(
+    user,
+    "home/events",
+    () =>
+      prisma.event
+        .findMany({
+          where: {
+            startDatetime: {
+              gt: new Date(),
+            },
+          },
+          orderBy: {
+            startDatetime: "asc",
+          },
+          take: 3,
+        })
+        .then((events) =>
+          events.map((event) => ({
+            ...event,
+            description: DOMPurify.sanitize(event.description),
+            descriptionEn: event.descriptionEn
+              ? DOMPurify.sanitize(event.descriptionEn)
+              : event.descriptionEn,
+          })),
+        ),
+    [CacheDependency.EVENTS],
+  );
 
   // MEETINGS
-  const upcomingMeetingPromise = prisma.meeting.findFirst({
-    where: {
-      date: {
-        gt: new Date(),
-      },
-    },
-    orderBy: {
-      date: "asc",
-    },
-  });
-  const previousMeetingPromise = prisma.meeting.findFirst({
-    where: {
-      date: {
-        lt: new Date(),
-      },
-    },
-    orderBy: {
-      date: "desc",
-    },
-  });
+  const upcomingMeetingPromise = userLevelCached(
+    user,
+    "home/upcomingMeetings",
+    () =>
+      prisma.meeting.findFirst({
+        where: {
+          date: {
+            gt: new Date(),
+          },
+        },
+        orderBy: {
+          date: "asc",
+        },
+      }),
+    [CacheDependency.MEETINGS],
+  );
+  const previousMeetingPromise = userLevelCached(
+    user,
+    "home/previousMeetings",
+    () =>
+      prisma.meeting.findFirst({
+        where: {
+          date: {
+            lt: new Date(),
+          },
+        },
+        orderBy: {
+          date: "desc",
+        },
+      }),
+    [CacheDependency.MEETINGS],
+  );
 
   // CAFE OPENING TIMES
-  const cafeOpenPromise = prisma.markdown.findFirst({
-    where: {
-      name: `cafe:open:${new Date().getDay() - 1}`, // we assign monday to 0, not sunday
-    },
-  });
+  const cafeOpenPromise = globallyCached(
+    "cafeOpenTimesToday",
+    () =>
+      authorizedPrismaClient.markdown.findFirst({
+        where: {
+          name: `cafe:open:${new Date().getDay() - 1}`, // we assign monday to 0, not sunday
+        },
+      }),
+    [CacheDependency.CAFE_OPEN_TIMES],
+  );
 
   // COMMIT DATA
-  const commitPromise = fetch("/api/home").then((res) => res.json());
+  const commitPromise = globallyCached("commitData", async () => {
+    const res = await fetch("/api/home");
```

Comment:
The `/api/home` endpoint already caches the commit data to avoid getting rate limited by Github, should probably extract the logic and remove the endpoint entirely